src/parser/Parser.ts

```diff
@@ -902,14 +897,14 @@ export class Parser {
                     match(Lexeme.Newline);
                 }
             } else {
-                let thenStatement = declaration(Lexeme.ElseIf, Lexeme.Else);
-                if (!thenStatement) {
+                let maybeThenBranch = block(Lexeme.Newline, Lexeme.Eof, Lexeme.ElseIf, Lexeme.Else);
```

Comment:
This is the core of the change.  Even in single-line `if` statements, we need to look for a `block()` instead of a single statement (via `declaration()`).  The remainder of the diff adjusts where statement separators are parsed (moved from the various functions that return statements and consolidated in `block()` and `declaration()`)